cypress/e2e/smoke/load_safe.cy.js

```diff
@@ -1,83 +1,106 @@
-    cy.get('[id=":r1:"]').type('RandomText')
-    cy.get('[id=":r1:-label"]').contains(INVALID_INPUT_ERROR_MSG)
-    cy.get('[id=":r1:"]').clear().type(NON_CONTRACT_ADDRESS)
-    cy.get('[id=":r1:-label"]').contains(INVALID_ADDRESS_ERROR_MSG)
-    //ENS names are not working in localhost
-    //cy.get('[id=":r1:"]')
-    //   .clear()
-    //   .type(SAFE_ENS_NAME)
-    //   .should('have.value', SAFE_ENS_NAME_TRANSLATED)
+    // Selects Rinkeby
+    cy.contains('Select network on which the Safe was created:').contains('span', 'Ethereum').click()
+    cy.get('ul li').contains('Rinkeby').click()
+    cy.contains('Select network on which the Safe was created:').contains('span', 'Rinkeby')
+
+    cy.contains('Continue').click()
   })
 
-  it('Should enter a valid QRCode', () => {
+  it('should accept name the safe', () => {
+    // alias the address input label
+    cy.get('input[name="address"]').parent().prev('label').as('addressLabel')
+
+    // Name input should have a placeholder ending in 'rinkeby-safe'
+    cy.get('input[name="name"]').should('have.attr', 'placeholder').should('contain', 'rinkeby-safe')
+    // Input a custom name
+    cy.get('input[name="name"]').type('Test safe name').should('have.value', 'Test safe name')
+
+    // Input incorrect Safe address
+    cy.get('input[name="address"]').type('RandomText')
+    cy.get('@addressLabel').contains(INVALID_INPUT_ERROR_MSG)
+
+    // Type an invalid address
+    cy.get('input[name="address"]').clear().type(EOA_ADDRESS)
+    cy.get('@addressLabel').contains(INVALID_ADDRESS_ERROR_MSG)
+
+    // Type a ENS name
+    cy.get('input[name="address"]').clear().type(SAFE_ENS_NAME)
+    // giving time to the ENS name to be translated
+    cy.get('input[name="address"]', { timeout: 10000 }).should('have.value', SAFE_ENS_NAME_TRANSLATED)
+
+    // Uploading a QR code
     cy.findByTestId('QrCodeIcon').click()
-    cy.findByText('Upload an image').click()
+    cy.contains('Upload an image').click()
     cy.get('[type="file"]').attachFile('../fixtures/rinkeby_safe_QR.png')
-    cy.get('[id=":r1:"]').should('have.value', SAFE_QR_CODE_ADDRESS)
-    cy.findByText('Continue').click()
+
+    // The address field should be filled with the QR code's address
+    cy.get('input[name="address"]').should('have.value', SAFE_QR_CODE_ADDRESS)
+    cy.contains('Continue').click()
   })
 
-  it('Should name the 1st owner', () => {
-    // Owners
-    // ENS names as default names have been removed in #376
-    //cy.findByPlaceholderText(OWNER_ENS_DEFAULT_NAME).parents('div[data-testid="owner-row"]').findByText(OWNER_ADDRESS)
-    cy.get('[name="owners.0.name"]').type('Test Owner Name').should('have.value', 'Test Owner Name')
-    cy.findByText('Continue').click()
+  it('should resolve ENS names for Safe owners', () => {
+    // Finds ENS name as one of the owners (give some time to the resolver)
+    cy.findByPlaceholderText(OWNER_ENS_DEFAULT_NAME, { timeout: 20000 })
+      .parents('.MuiGrid-container')
```

Comment:
Is this grid selector really needed?

Reply:
I found it the less likely classname to change given that the owner name and address are inside of a grid
<img width="508" alt="Screenshot 2022-09-21 at 16 36 58" src="https://user-images.githubusercontent.com/32431609/191533765-a32fd141-192d-4cdb-94d8-1de17f9a179b.png">